PayPalNative/src/main/java/com/braintreepayments/api/PayPalNativeClient.java

```diff
@@ -50,33 +79,130 @@ public void onResult(Configuration configuration, Exception error) {
     }
 
-    private void startPayPalNativeCheckout(final FragmentActivity activity, Configuration configuration, String payPalClientId, final PayPalNativeCheckoutRequest payPalRequest, final PayPalNativeFlowStartedCallback callback) {
-        // TODO: configure paypal native checkout
-
+    private void startPayPalNativeCheckout(final FragmentActivity activity, final Configuration configuration, final String payPalClientId, final PayPalNativeCheckoutRequest payPalRequest, final PayPalNativeFlowCallback callback) {
         internalPayPalClient.sendRequest(activity, payPalRequest, new PayPalInternalClientCallback() {
             @Override
-            public void onResult(PayPalResponse payPalResponse, Exception error) {
+            public void onResult(final PayPalResponse payPalResponse, final Exception error) {
                 if (payPalResponse != null) {
-                    // TODO: start native checkout
-                    callback.onResult(new BraintreeException("TODO: integrate native checkout with Braintree SDK."));
+                    startNativeCheckout(activity, configuration, payPalClientId, payPalResponse, callback);
                 } else {
-                    callback.onResult(error);
+                    callback.onResult(null, error);
                 }
             }
         });
     }
 
+    private void startNativeCheckout(final FragmentActivity activity, final Configuration configuration, final String clientId, final PayPalResponse payPalResponse, final PayPalNativeFlowCallback callback) {
+        final String redirectUrl = getPayPalReturnUrl(activity);
+        Environment environment = Environment.SANDBOX;
+        if ("production".equalsIgnoreCase(configuration.getEnvironment())) {
+            environment = Environment.LIVE;
+        }
+
+        CheckoutConfig checkoutConfig = new CheckoutConfig(
+                activity.getApplication(),
+                clientId,
+                environment,
+                redirectUrl
+        );
+        PayPalCheckout.setConfig(checkoutConfig);
+        PayPalCheckout.start(
+                new CreateOrder() {
+                    @Override
+                    public void create(final CreateOrderActions createOrderActions) {
+                        createOrderActions.set(payPalResponse.getPairingId());
+                    }
+                },
+                new OnApprove() {
+                    @Override
+                    public void onApprove(final Approval approval) {
+                        braintreeClient.sendAnalyticsEvent("paypal.native.approved");
+                        approved(approval, payPalResponse, callback);
+                    }
+                },
+                new OnCancel() {
+                    @Override
+                    public void onCancel() {
+                        braintreeClient.sendAnalyticsEvent("paypal.native.client_cancel");
+                        callback.onResult(null, new BraintreeException("User Cancelled"));
```

Comment:
```suggestion
                        callback.onResult(null, new BraintreeException("User Canceled PayPal"));
```
Nitpick - but aligning this with the legacy cancellation message